lib/lpc55-rot-startup/src/lib.rs

```diff
@@ -75,6 +75,38 @@ fn puf_check(puf: &lpc55_pac::PUF) {
     }
 }
 
+fn enable_debug(peripherals: &lpc55_pac::Peripherals) {
+    let beacon = peripherals.SYSCON.debug_auth_beacon.read().bits();
+
+    if beacon != 0 {
```

Comment:
RFD333 said "non-zero" but I don't know if we want to consider making this more specific. I wasn't sure how to combine that with the credentials for choosing A/B images.

Reply:
Can you provide a link to the relevant section of the RFD? I'm not able to find the reference. Generally, based on my current understanding I'd agree though: this does need to be more specific.

AFAIK the beacon has two parts: the lower half of the register is the `credential beacon`. This is set in the debug credential at time of creation and is part of the data that is signed by one of the trust anchors. This allows the issuer to limit the powers of the credential. We use this value in the bootleby image selection here: https://github.com/oxidecomputer/bootleby/blob/main/src/bin/bootleby.rs#L154. The other half of the register (aka the auth beacon) can be set to an arbitrary value by the holder of the debug auth credential as part of the protocol exchange with the ROM. This allows the credential holder to provide some input (the auth beacon) that is limited in scope by the credential beacon.

Using bootleby as an example: The credential beacon that allows the debug auth credential holder to control bootleby boot selection is `18578`. I think of this as defining the scope of the credential: in this case it's overriding the default A / B image selection. This is to say that in order to control bootleby image selection the debug auth credential must have the `CB` field set to `18578`. During the protocol exchange with the ROM the A / B image is selected by setting the authentication beacon to either `0` to select image A, or `1` to select image B. These values are documented in OANA: https://github.com/oxidecomputer/oana#beacon

If I'm understanding all of this correctly the code here will allow the holder of any debug auth credential to debug the chip via SWD. This includes the credentials issued for overriding the bootleby image selection. This probably isn't what we want. Instead I think we need to add another credential beacon value to OANA and check that the provided debug auth credential has this credential beacon set. I don't see any use for the auth beacon here but it's a mechanism available if we need it.

Reply:
Thanks for the reference. I searched for the quoted text ("non-zero") and came up empty handed. I think it's far more likely that we'll want a test / mfg system that can override boot selection to ensure that both A and B images can boot but that *cannot* do generic SWD things. If we do run into a situation where we want a credential that's able to do both things we could probably just use two separate credentials?

Reply:
I chose a number and put it in the OANA and double checked that setting the cred beacon _will_ unlock debug but anything will _not_.